elements/lisk-p2p/src/p2p_types.ts

```diff
@@ -65,6 +65,13 @@ export interface P2PPeerInfo {
 	readonly internalState?: P2PInternalState;
 }
 
+export type P2PEnhancedPeerInfo = {
+	readonly dateAdded?: Date;
```

Comment:
does this all needs to be optional? can we initialize it initially?

Reply:
It's optional because some of the properties such as `sourceAddress` are initialized in base peer class during fetching and others (e.g. `dateAdded`, `numOfConnectionFailures`) cannot initialized until it `peerInfo` reaches peerBook.